webview/WebView2_spec.md

```diff
@@ -0,0 +1,247 @@
+    }
+}
+```
+
+<br>
+
+**Reload:**
+Fires navigation event without altering the navigation stack.
+
+<br>
+
+# Remarks
+
+// This API may ship before the WinRT CoreWebView object is included, meaning functionality of the Webview2 will be limited to the APIs included on this object.
+
+# API Notes
+
+**Properties:**
+<br>
+Source – Gets or sets the Uniform Resource Identifier (URI) source of the HTML content to display in the WebView2 control.
+
+CoreWebView2 – Stores the core WebView object with full access to Anaheim WebView API set.
+<br>
+
+**Methods:**
+<br>
+void NavigateToString(String htmlContent) – Loads the specified HTML content as a new document.
+
+Windows.Foundation.IAsyncOperation ExecuteScriptAsync(String javascriptCode) –
+
+Executes the specified script function from the currently loaded HTML, with specific arguments, as an asynchronous action.
+
+void Reload() – Reloads the current content in the WebView2.
+
+void Stop() – Stops the current WebView2 navigation or download.
+<br>
+
+**Events:**
+<br>
+WebMessageReceived – Occurs when the content contained in the WebView2 control passes a string to the application by using JavaScipt.
+
+NavigationStarting – Occurs before the WebView2 navigates to new content. You can cancel navigation in a handler for this event by setting the WebViewNavigationStartingEventArgs.Cancel property to true.
+
+ContentLoading - ContentLoading fires when new content has started loading on the webview's main frame or if a same page navigation occurs (such as through fragment navigations or history.pushState navigations). This follows the NavigationStarting event and precedes the NavigationCompleted event.
+
+SourceChanged - Occurs when the 'Source' property is changed
+
+NavigationCompleted – Occurs when the WebView2 has finished loading the current content or if the navigation has failed.
+
+<br>
+
+# API Details
+
+API Details
+
+```
+namespace MU\_XC\_NAMESPACE
+{
+    [WUXC\_VERSION\_PREVIEW]
+    [webhosthidden]
+
+    runtimeclass WebView2ElementNavigationCompletedEventArgs
+    {
+        Boolean IsSuccess{ get; };
+        Windows.Web.WebErrorStatus WebErrorStatus{ get; };
+    }
+
+    [WUXC\_VERSION\_PREVIEW]
+    [webhosthidden]
+    runtimeclass WebView2ElementWebMessageReceivedEventArgs
+    {
+        String Source{ get; };
+        String WebMessageAsJson{ get; };
+        String WebMessageAsString{ get; };
+    }
+
+    [WUXC\_VERSION\_PREVIEW]
+    [webhosthidden]
+    runtimeclass WebView2ElementNavigationStartingEventArgs
+    {
+        String Uri{ get; };
+        Boolean IsUserInitiated{ get; };
+        Boolean IsRedirected{ get; };
+        // RequestHeaders will be included once an appropriate winrt collection is available.
+        Boolean Cancel{ get; set; };
+    }
+
+    [WUXC\_VERSION\_PREVIEW]
+    [webhosthidden]
+    [MUX\_PROPERTY\_CHANGED\_CALLBACK(TRUE)]
```

Comment:
All of these [MUX attributes are implementation details, shouldn't be in the API